segfault in UcxEndpointTest, uct_rc_iface_tx_ops_cleanup

http://hpc-master.lab.mtl.com:8080/blue/rest/organizations/jenkins/pipelines/ucx/runs/2692/nodes/108/steps/1102/log/?start=0
```
[2019-11-16T17:22:56.997Z] Running org.openucx.jucx.UcpEndpointTest
[2019-11-16T17:22:58.890Z] #
[2019-11-16T17:22:58.890Z] # A fatal error has been detected by the Java Runtime Environment:
[2019-11-16T17:22:58.890Z] #
[2019-11-16T17:22:58.890Z] #  SIGSEGV (0xb) at pc=0x00007fd407bdb470, pid=8916, tid=0x00007fd42b834700
[2019-11-16T17:22:58.890Z] #
[2019-11-16T17:22:58.890Z] # JRE version: Java(TM) SE Runtime Environment (8.0_121-b13) (build 1.8.0_121-b13)
[2019-11-16T17:22:58.890Z] # Java VM: Java HotSpot(TM) 64-Bit Server VM (25.121-b13 mixed mode linux-amd64 compressed oops)
[2019-11-16T17:22:58.890Z] # Problematic frame:
[2019-11-16T17:22:58.890Z] # C  [libuct_ib.so+0x15470]  uct_rc_iface_tx_ops_cleanup+0x20
[2019-11-16T17:22:58.890Z] #
[2019-11-16T17:22:58.890Z] # Core dump written. Default location: /scrap/jenkins/workspace/ucx-2/build-test/bindings/java/core or core.8916
[2019-11-16T17:22:58.890Z] #
[2019-11-16T17:22:58.890Z] # An error report file with more information is saved as:
[2019-11-16T17:22:58.890Z] # /scrap/jenkins/workspace/ucx-2/build-test/bindings/java/hs_err_pid8916.log
[2019-11-16T17:22:58.890Z] #
[2019-11-16T17:22:58.890Z] # If you would like to submit a bug report, please visit:
[2019-11-16T17:22:58.890Z] #   http://bugreport.java.com/bugreport/crash.jsp
[2019-11-16T17:22:58.890Z] # The crash happened outside the Java Virtual Machine in native code.
[2019-11-16T17:22:58.890Z] # See problematic frame for where to report the bug.
[2019-11-16T17:22:58.890Z] #
[2019-11-16T17:22:58.890Z] /bin/sh: line 1:  8916 Aborted                 (core dumped) /hpc/local/oss/java/jdk1.8.0_121/jre/bin/java -jar /scrap/jenkins/workspace/ucx-2/build-test/bindings/java/src/main/native/build-java/surefire/surefirebooter1345070732321084074.jar /scrap/jenkins/workspace/ucx-2/build-test/bindings/java/src/main/native/build-java/surefire/surefire731707726494747762tmp /scrap/jenkins/workspace/ucx-2/build-test/bindings/java/src/main/native/build-java/surefire/surefire_04665215650139385498tmp
[2019-11-16T17:22:58.890Z] 
[2019-11-16T17:22:58.890Z] Results :
[2019-11-16T17:22:58.890Z] 
[2019-11-16T17:22:58.890Z] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
[2019-11-16T17:22:58.890Z] 
[2019-11-16T17:22:58.890Z] [INFO] ------------------------------------------------------------------------
[2019-11-16T17:22:58.890Z] [INFO] BUILD FAILURE
[2019-11-16T17:22:58.890Z] [INFO] ------------------------------------------------------------------------
[2019-11-16T17:22:58.890Z] [INFO] Total time: 01:47 min (Wall Clock)
[2019-11-16T17:22:58.890Z] [INFO] Finished at: 2019-11-16T19:22:58+02:00
[2019-11-16T17:22:59.452Z] [INFO] Final Memory: 29M/346M
[2019-11-16T17:22:59.452Z] [INFO] ------------------------------------------------------------------------
[2019-11-16T17:22:59.453Z] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.12.4:test (default-test) on project jucx: Execution default-test of goal org.apache.maven.plugins:maven-surefire-plugin:2.12.4:test failed: The forked VM terminated without saying properly goodbye. VM crash or System.exit called ? -> [Help 1]
```

--- a/bindings/java/src/test/java/org/openucx/jucx/UcpEndpointTest.java
+++ b/bindings/java/src/test/java/org/openucx/jucx/UcpEndpointTest.java
@@ -300,7 +300,14 @@ public class UcpEndpointTest {
         }
 
         assertTrue(success.get());
+
         progressThread.interrupt();
+        try {
+            progressThread.join();
+        } catch (InterruptedException e) {
+
+        }
+
         ep.close();
         worker2.close();
         worker1.close();

